### PR TITLE
ci: setup CodeQL code scanning

### PR DESCRIPTION
**Pull request by ylemkimon**
https://securitylab.github.com/tools/codeql

### CODE REVIEW

**Comment by kevinbarabash** [comment]
CodeQL looks like a very useful tool.  Thanks for setting it up.  I'd like to experiment with in some of my repos. 🙂

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '0 0 * * 1'
```
Comment (kevinbarabash) [inline .github/workflows/codeql-analysis.yml:9]: Why not run it on every pull request?  I'm not familiar with this tool so I don't know what the best practices are.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (ylemkimon) [inline .github/workflows/codeql-analysis.yml:7]: It does run on every pull request.

Comment (kevinbarabash) [reply .github/workflows/codeql-analysis.yml:7]: I see.  The `branches` specifies which branch(es) the pull request is against.

**Review by kevinbarabash — APPROVED** [review]
LGTM

**Pull request merged**